### PR TITLE
Example using :scope pseudo-class - Fixes #6676

### LINKED ISSUE #6676: Content suggestion: Example of :scope in element.querySelecotr()
## What is the new suggestion?
What if among examples of `element.querySelector()`
https://developer.mozilla.org/en-US/docs/Web/API/element/querySelector
would be something with `:scope` in selectors?
https://developer.mozilla.org/en-US/docs/Web/CSS/:scope

## Why is it important or useful?
It seems relevant, considering there aren’t any other uses for `:scope` now, and it’s basically the only place it matters at the moment.

### PR DESCRIPTION
**Pull request by Anurella**
> What was wrong/why is this fix needed? (quick summary only)
An example was needed to highlight the use of :scope pseudo class.

> Issue number (if there is an associated issue)
Fixes #6676

> Anything else that could help us review it

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,7 +213,7 @@ and use it to make a pull request.
 1. Next, you'll want to start the local preview service, so you can see
 the changes you'll make as they would look in production. Once started,
 this local preview service is available at `http://localhost:5000`
-within your browser.
+within your browser. Do not forget to run npm install for dependencies.
```
Comment (Ryuno-Ki) [inline README.md:216]:
```suggestion
within your browser. Do not forget to run `npm install` for dependencies.
```
By declaring it as `<code>`, it is easier to understand what to run.

File: README.md [hunk 2]
```diff
@@ -213,7 +213,7 @@ and use it to make a pull request.
 1. Next, you'll want to start the local preview service, so you can see
 the changes you'll make as they would look in production. Once started,
 this local preview service is available at `http://localhost:5000`
```
Comment (Ryuno-Ki) [inline README.md:215]:
```suggestion
this local preview service is available at `http://localhost:5000/`
```
can save a redirect in the web browser (tiny performance win).

File: changes.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## Issues to be worked on
+
+1. [Example of :scope in element.querySelector()](https://github.com/mdn/content/issues/6676)
```
Comment (Ryuno-Ki) [inline changes.md:3]: What is the motivation for this file?

File: files/en-us/web/api/element/queryselector/index.html [hunk 4]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
 <pre class="brush:js">var el = document.body.querySelector("style[type='text/css'], style:not([type])");
 </pre>
 
+
+
+<h3 id="Get_direct_descendants_using_scope_pseudo_class"> Get direct descendants using :scope pseudo-class</h3>
+<p> This example uses the <a href="/en-US/docs/Web/CSS/:scope">:scope</a> pseudo-class to 
+  retrieve direct children of the <code>parentElement </code></p>
```
Comment (Ryuno-Ki) [inline files/en-us/web/api/element/queryselector/index.html:80]:
```suggestion
  retrieve direct children of the <code>parentElement</code>.</p>
```

File: files/en-us/web/api/element/queryselector/index.html [hunk 5]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
...
+    display:block;
+    margin-bottom: 5px;
+  }
+  .red span {
+    background-color: red;
+    padding:5px;
+  }
+</pre>
+
+<h4 id="Javascript"> JavaScript</h4>
```
Comment (Ryuno-Ki) [inline files/en-us/web/api/element/queryselector/index.html:109]:
```suggestion
<h4 id="Javascript">JavaScript</h4>
```

File: files/en-us/web/api/element/queryselector/index.html [hunk 6]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
...
+<pre class="brush: js">
+  const parentElement = document.querySelector('#parent');
+  let allChildren = parentElement.querySelectorAll(":scope > span");
+  allChildren.forEach(item => item.classList.add("red"));
+</pre>
+
+<h4 id="result"> Result </h4>
```
Comment (Ryuno-Ki) [inline files/en-us/web/api/element/queryselector/index.html:116]:
```suggestion
<h4 id="result">Result</h4>
```

**Review by Ryuno-Ki — COMMENTED** [review]
I believe, the `package-lock.json` should not be part of this PR.

I've highlighted some small nitpicks you can address while waiting for the proper review by the team.

File: changes.md (hunk 3 above)
Comment (Anurella) [reply changes.md:3]: Added to .gitignore

**Comment by Ryuno-Ki** [comment]
Looks good to me.

Whoever happens to review this, please take a look, whether the id attributes have the right casing.
(I'm unsure what the right way is off-hand).

**Comment by Ryuno-Ki** [comment]
The failing tests is about `changes.md`:

> First line in a file should be a top-level heading [Context: "## Issues to be worked on"]

**Comment by Ryuno-Ki** [comment]
I filled a follow-up to deal with the detected flaws, so don't worry about them.

File: files/en-us/web/api/element/queryselector/index.html [hunk 7]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
 <pre class="brush:js">var el = document.body.querySelector("style[type='text/css'], style:not([type])");
 </pre>
 
+
+
```
Comment (sideshowbarker) [inline files/en-us/web/api/element/queryselector/index.html:77]:
```suggestion


```

File: files/en-us/web/api/element/queryselector/index.html [hunk 8]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
...
+<h3 id="Get_direct_descendants_using_scope_pseudo_class"> Get direct descendants using :scope pseudo-class</h3>
```
Comment (sideshowbarker) [inline files/en-us/web/api/element/queryselector/index.html:78]:
```suggestion
<h3 id="get_direct_descendants_using_the_scope_pseudo_class">Get direct descendants using the :scope pseudo-class</h3>
```

File: files/en-us/web/api/element/queryselector/index.html [hunk 9]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
...
+<p> This example uses the <a href="/en-US/docs/Web/CSS/:scope">:scope</a> pseudo-class to 
+  retrieve direct children of the <code>parentElement</code>.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/element/queryselector/index.html:80]:
```suggestion
<p>This example uses the <a href="/en-US/docs/Web/CSS/:scope">:scope</a> pseudo-class to retrieve direct children of the <code>parentElement</code> element.</p>
```

File: files/en-us/web/api/element/queryselector/index.html [hunk 10]
```diff
@@ -73,6 +73,51 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
...
+<pre class="brush: js">
+  const parentElement = document.querySelector('#parent');
+  let allChildren = parentElement.querySelectorAll(":scope > span");
+  allChildren.forEach(item => item.classList.add("red"));
+</pre>
+
+<h4 id="result">Result</h4>
+<p>{{ EmbedLiveSample('Get_direct_descendants_using_scope_pseudo_class', 600, 160) }}</p>
+
+
```
Comment (sideshowbarker) [inline files/en-us/web/api/element/queryselector/index.html:119]:
```suggestion


```

**Review by sideshowbarker — CHANGES_REQUESTED** [review]
Looks great — I just have a few styling changes to suggest

**Comment by sideshowbarker** [comment]
Would also be good to have a review from @rachelandrew

**Comment by sideshowbarker** [comment]
@Anurella This branch originally had another commit, “Added a note and example for using fill() with empty array” 443067fc6a0d184bc2f28b3d23a06d823d1ab125 that I think you didn’t intend to be part of this PR but instead for another PR? I have stripped that commit out of this PR branch, so if you don’t have a local copy of that commit, here’s copy of the diff:

````diff
commit 443067fc6a0d184bc2f28b3d23a06d823d1ab125
Author: Anurella <ndukwuamaka@yahoo.com>
Date:   Wed Jul 28 22:11:09 2021 +0100

    Added a note and example for using fill() with empty array

diff --git a/files/en-us/web/javascript/reference/global_objects/array/fill/index.md b/files/en-us/web/javascript/reference/global_objects/array/fill/index.md
index e42dae9481..ae6f2a2594 100644
--- a/files/en-us/web/javascript/reference/global_objects/array/fill/index.md
+++ b/files/en-us/web/javascript/reference/global_objects/array/fill/index.md
@@ -48,6 +48,11 @@ The modified array, filled with `value`.
 - `fill` is a mutator method: it will change the array itself and return it, not a copy of it.
 - If the first parameter is an object, each slot in the array will reference that object.
 
+<div class="note notecard" markdown="1">
+Note: Using Array.Fill() on an empty array would not modify it as the array has nothing to be modified. To use Array.Fill() when declaring an array, make sure to assign slots to the array.
+<a href="#using_fill_to_populate_an_empty_array">See example</a>
+</div>
+
 ## Polyfill
 
 ```js
@@ -135,6 +140,15 @@ console.log(arr[1][0]); // 1
 console.log(arr[2][0]); // 1
 ```
 
+### Using fill() to populate an Empty Array
+
+This example shows how to populate an array with random values. The end value 
+does not have to be specified.
+
+```js
+let tempGirls = Array(5).fill("girl",0);
+```
+
 ## Specifications
 
 {{Specifications}}
````

File: files/en-us/web/api/element/queryselector/index.html [hunk 11]
```diff
@@ -73,6 +73,50 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
 <pre class="brush:js">var el = document.body.querySelector("style[type='text/css'], style:not([type])");
 </pre>
 
+
+
```
Comment (sideshowbarker) [inline files/en-us/web/api/element/queryselector/index.html:77]:
```suggestion
```

File: files/en-us/web/api/element/queryselector/index.html [hunk 12]
```diff
@@ -73,6 +73,48 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
...
+  const parentElement = document.querySelector('#parent');
+  let allChildren = parentElement.querySelectorAll(":scope > span");
+  allChildren.forEach(item => item.classList.add("red"));
+</pre>
+
+<h4 id="result">Result</h4>
+<p>{{ EmbedLiveSample('Get_direct_descendants_using_scope_pseudo_class', 600, 160) }}</p>
+
+
+
```
Comment (sideshowbarker) [inline files/en-us/web/api/element/queryselector/index.html:117]:
```suggestion

```

File: files/en-us/web/api/element/queryselector/index.html [hunk 13]
```diff
@@ -73,6 +73,45 @@ <h3 id="Find_a_specific_element_with_specific_values_of_an_attribute">Find a spe
...
+<h4 id="Javascript">JavaScript</h4>
+<pre class="brush: js">
+  const parentElement = document.querySelector('#parent');
+  let allChildren = parentElement.querySelectorAll(":scope > span");
+  allChildren.forEach(item => item.classList.add("red"));
+</pre>
+
+<h4 id="result">Result</h4>
+<p>{{ EmbedLiveSample('Get_direct_descendants_using_scope_pseudo_class', 600, 160) }}</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/element/queryselector/index.html:114]:
```suggestion
<p>{{ EmbedLiveSample('Get_direct_descendants_using_scope_pseudo_class', 600, 160) }}</p>

```

**Review by sideshowbarker — APPROVED** [review]

**Review by rachelandrew — APPROVED** [review]

**Pull request merged**